mirgecom/boundary.py

```diff
@@ -51,74 +60,379 @@ class FluidBoundary(metaclass=ABCMeta):
+    .. automethod:: temperature_gradient_flux
     """
 
     @abstractmethod
-    def inviscid_divergence_flux(self, discr, btag, eos, cv_minus, dv_minus,
-                                 **kwargs):
-        """Get the inviscid boundary flux for the divergence operator."""
-        pass
+    def inviscid_divergence_flux(self, discr, btag, gas_model, state_minus,
+                                 numerical_flux_func, **kwargs):
+        """Get the inviscid boundary flux for the divergence operator.
+
+        This routine returns the facial flux used in the divergence
+        of the inviscid fluid transport flux.
+
+        Parameters
+        ----------
+        discr: :class:`~grudge.eager.EagerDGDiscretization`
+
+            A discretization collection encapsulating the DG elements
+
+        state_minus: :class:`~mirgecom.gas_model.FluidState`
+
+            Fluid state object with the conserved state, and dependent
+            quantities for the (-) side of the boundary specified by
+            *btag*.
+
+        gas_model: :class:`~mirgecom.gas_model.GasModel`
+
+            Physical gas model including equation of state, transport,
+            and kinetic properties as required by fluid state
+
+        numerical_flux_func:
+
+            Function should return the numerical flux corresponding to
+            the divergence of the inviscid transport flux. This function
+            is typically backed by an approximate Riemann solver, such as
+            :func:`~mirgecom.inviscid.inviscid_facial_flux_rusanov`.
+
+        Returns
+        -------
+        :class:`mirgecom.fluid.ConservedVars`
+        """
+
+    @abstractmethod
+    def viscous_divergence_flux(self, discr, btag, gas_model, state_minus,
+                                grad_cv_minus, grad_t_minus, **kwargs):
+        """Get the viscous boundary flux for the divergence operator.
+
+        This routine returns the facial flux used in the divergence
+        of the viscous fluid transport flux.
+
+        Parameters
+        ----------
+        discr: :class:`~grudge.eager.EagerDGDiscretization`
+
+            A discretization collection encapsulating the DG elements
+
+        state_minus: :class:`~mirgecom.gas_model.FluidState`
+
+            Fluid state object with the conserved state, and dependent
+            quantities for the (-) side of the boundary specified
+            by *btag*.
+
+        grad_cv_minus: :class:`~mirgecom.fluid.ConservedVars`
+
+            The gradient of the conserved quantities on the (-) side
+            of the boundary specified by *btag*.
+
+        grad_t_minus: numpy.ndarray
+
+            The gradient of the fluid temperature on the (-) side
+            of the boundary specified by *btag*.
+
+        gas_model: :class:`~mirgecom.gas_model.GasModel`
+
+            Physical gas model including equation of state, transport,
+            and kinetic properties as required by fluid state
+
+        numerical_flux_func:
```

Comment:
same as above.  complete the doc string.  Is keyword `numerical_flux_func` required?  if so, add it above.